cmd/state-installer/internal/installer/installation.go

```diff
@@ -22,7 +23,17 @@ func backupFiles(targetFiles []string) ([]string, error) {
 	for _, t := range targetFiles {
 		if fileutils.TargetExists(t) {
 			newName := fmt.Sprintf("%s.bak", t)
-			err := os.Rename(t, newName)
+			if fileutils.TargetExists(newName) {
+				_ = os.Remove(newName)
+			}
+			var err error
+			if runtime.GOOS == "windows" {
+				// Note: We could also rename the files here, in which case the installation would always pass on Windows (even if the executable is still running), but we would not be able to remove the .bak file after the installation (if the executable is still running).
+				// We opted for this solution, as it is more compliant with other Windows installations, where the installation requires all existing programmes to be stopped prior to updates.
```

Comment:
I'd rather we try our best effort to make things work consistently across platforms, which means that if we can make it work while the executable is running we should.

People being able to use tooling without issues is more important than a few stale backup files that we can clean up on a later call.